### PR TITLE
Bump sdk 0451

### PR DESCRIPTION
**Pull request by yaruwangway**
- bump cosmos-sdk to 0.45.1
- bump go version to 1.17

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,5 +38,5 @@ The following example shows routing from Terra to Osmosis through the Hub:
 # References
 
 - https://www.mintscan.io/cosmos/proposals/56
-- https://github.com/cosmos/ibc-go/pull/373
+- https://github.com/cosmos/ibc-go/v3/pull/373
```
Comment (okwme) [inline README.md:41]:
```suggestion
- https://github.com/cosmos/ibc-go/pull/373
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -52,7 +52,7 @@ taken place in a github issue, that PR runs a high likelihood of being rejected.
 Other notes:
 
 - Looking for a good place to start contributing? How about checking out some
-  [good first issues](https://github.com/cosmos/ibc-go/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)
+  [good first issues](https://github.com/cosmos/ibc-go/v3/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)
```
Comment (okwme) [inline CONTRIBUTING.md:55]:
```suggestion
  [good first issues](https://github.com/cosmos/ibc-go/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)
```

**Comment by jtieri** [comment]
Everything looks good but I notice two proposed changes that I am unsure on. I would think we want the existing line in this case e.g. `option go_package = "github.com/strangelove-ventures/packet-forward-middleware/router/types";`

As it's my understanding this line is defining the import path of the package which will contain all the generated code for these two files, so to me it makes more sense that this points to the types in the actual packet-forward-middleware.

Could either of you confirm this before i resolve conflicts and commit merge? @yaruwangway @okwme 

in `genesis.proto`
```
<<<<<<< bump-sdk-0451
option go_package = "github.com/cosmos/ibc-go/v3/modules/apps/router/types";
=======
option go_package = "github.com/strangelove-ventures/packet-forward-middleware/router/types";
>>>>>>> main
```

in `query.proto`
```
<<<<<<< bump-sdk-0451
option go_package = "github.com/cosmos/ibc-go/v3/modules/apps/router/types";
=======
option go_package = "github.com/strangelove-ventures/packet-forward-middleware/router/types";
>>>>>>> main
```

**Comment by yaruwangway** [comment]
hi @jtieri  it makes sense to choose main about `option go_package...`!
just  my pr does not contain https://github.com/strangelove-ventures/packet-forward-middleware/pull/4/files, so directly merge might cover this pr. maybe i rebase before you merge ?

**Comment by okwme** [comment]
thanks @jtieri ! Will you be tagging a new release for this?

**Pull request merged**